src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ExportsFileWriter.cs

```diff
@@ -34,20 +36,58 @@ public void EmitExportedMethods()
                 if (_context.Target.IsWindows)
                 {
                     streamWriter.WriteLine("EXPORTS");
+                    foreach (string symbol in _exportSymbols)
+                    {
+                        streamWriter.WriteLine($"   {symbol.Replace(',', ' ')}");
+                    }
                     foreach (var method in _methods)
+                    {
                         streamWriter.WriteLine($"   {method.GetUnmanagedCallersOnlyExportName()}");
+                    }
                 }
-                else if(_context.Target.IsOSXLike)
+                else if (_context.Target.IsOSXLike)
                 {
+                    foreach (string symbol in _exportSymbols)
+                    {
+                        string export;
+                        int comma = symbol.IndexOf(',');
```

Comment:
I am not sure whether it is better to strip everything after comma like this; or whether it would be better to pass the symbol with the `DATA` modifier in the .targets file on Windows only.

The `DATA` modifier is Windows-specific, so the current version is leaking Windows specific behavior to other platforms. On the other hand, modifiers like this can be useful on other platforms in future.

@MichalStrehovsky What do you think?

Reply:
I went back and forth on this too. The ILC already formats the def/version script differently depending on the platform so it is already platform aware but the code ends up simpler to pass the Windows specific DATA in the .targets file.

Reply:
I don't have a strong opinion either. This will work.

Reply:
I'm now feeling like the Windows difference should be in the .targets file like Jan suggested earlier in this thread.  The export writer code would be a lot simpler.